docs/quickstart/clustering.md

```diff
@@ -0,0 +1,129 @@
+
+Let's generate some sample data with 5 clusters; note that in most real-world use cases, you won't have ground truth data labels (which cluster a given observation belongs to). However, in this case, the ground truth data is available, which will help us explain the concepts more clearly.
+
+```{code-cell} ipython3
+from sklearn.datasets import make_blobs
+
+X, y = make_blobs(n_samples=1000,
+                  centers=5,
+                  n_features=20,
+                  random_state=0,
+                  cluster_std=3,
+                  center_box=(-10, 10))
+```
+
+## Visualizing clusters
+
+Visualizing high-dimensional data is difficult. A common approach is to reduce its dimensionality using PCA; this losses some information but can help us visualize the clusters. Let's run PCA on our data and plot it:
+
+```{code-cell} ipython3
+import matplotlib.pyplot as plt
+from sklearn.decomposition import PCA
+
+pca = PCA(n_components=2)
+X_pca = pca.fit_transform(X)
+
+plt.scatter(X_pca[:, 0], X_pca[:, 1], c=y)
+plt.grid()
+_ = plt.title("Data in PCA space")
+```
+
+We can see the clusters in our synthetic data. However, the clusters won't be as transparent when using real-world datasets as in our example dataset.
+
++++
+
+## Evaluation metrics
+
+When clustering data, we want to find the number of clusters that better fit the data. Most models have `n_clusters` as a parameter, so we have to try different values and evaluate which number is the best. To find the *best model*, we need to quantify the quality of the clusters. Here are three metrics you can use that do not require ground truth data:
+
+- `silhouette_score`: goes from -1 to +1, **higher is better** defined clusters  ([documentation](https://scikit-learn.org/stable/modules/clustering.html#silhouette-coefficient))
+- `calinski_harabasz_score`: a ratio, **higher is better** ([documentation](https://scikit-learn.org/stable/modules/clustering.html#calinski-harabasz-index))
+- `davies_bouldin_score`: **lower is better**, minimum value is 0 ([documentation](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.davies_bouldin_score.html#sklearn.metrics.davies_bouldin_score))
+
+Let's run a `KMeans` algorithm with different `n_clusters` and compute all three metrics; we'll highlight the metric value that best fits the data:
+
+```{code-cell} ipython3
+import pandas as pd
+from sklearn.cluster import KMeans
+from sklearn import metrics
+
+def score(X, n_clusters):
+    model = KMeans(n_init="auto", n_clusters=n_clusters, random_state=1)
+    model.fit(X)
+    predicted = model.predict(X)
+    return {"n_clusters": n_clusters,
+            "silhouette_score": metrics.silhouette_score(X, predicted),
+            "calinski_harabasz_score": metrics.calinski_harabasz_score(X, predicted),
+            "davies_bouldin_score": metrics.davies_bouldin_score(X, predicted)}
+
+
+df_metrics = pd.DataFrame(score(X, n_clusters) for n_clusters in (2, 3, 4, 5, 6, 7, 8, 9, 10))
```

Comment:
This line errors out to me on:
`TypeError: '<=' not supported between instances of 'str' and 'int'`
This is where I stopped since I cloudn't run it anymore or fix, please tag when fixed.

Reply:
Worth mentioning I converted the md to ipynb via jupysql

Reply:
i saw this before, I think you're running an old sklearn version. shar the full traceback. if that's the case, I can add a note saying sklearn>=1.2 is required

Reply:
>  Worth mentioning I converted the md to ipynb via jupysql

wdym? jupysql is a SQL client, it doesn't convert files.

Reply:
jupytext lol